src/patternfly/components/CodeEditor/code-editor.scss

```diff
@@ -0,0 +1,91 @@
+.pf-c-code-editor {
+  // control
+  --pf-c-code-editor__control--Height: calc(var(--pf-global--spacer--form-element) + var(--pf-global--spacer--form-element) + var(--pf-global--FontSize--md) * var(--pf-global--LineHeight--md) - var(--pf-global--BorderWidth--sm));
+
+  // action group
+  --pf-c-code-editor__action-group--ZIndex: var(--pf-global--ZIndex--xs);
+  --pf-c-code-editor__action-group--c-button--m-control--Color: var(--pf-global--Color--200);
+  --pf-c-code-editor__action-group--c-button--m-control--hover--Color: var(--pf-global--Color--100);
+  --pf-c-code-editor__action-group--c-button--m-control--focus--Color: var(--pf-global--Color--100);
+
+  // code
+  --pf-c-code-editor__code--before--BorderColor: var(--pf-global--BorderColor--100);
+  --pf-c-code-editor__code--before--BorderWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-code-editor__code--BackgroundColor: var(--pf-global--BackgroundColor--100);
+
+  // tab
+  --pf-c-code-editor__tab--BackgroundColor: var(--pf-global--BackgroundColor--100);
+  --pf-c-code-editor__tab--Color: var(--pf-global--Color--200);
+  --pf-c-code-editor__tab--ZIndex: var(--pf-global--ZIndex--xs);
+  --pf-c-code-editor__tab--Top: var(--pf-c-code-editor__code--before--BorderWidth);
+  --pf-c-code-editor__tab--PaddingTop: var(--pf-global--spacer--form-element);
+  --pf-c-code-editor__tab--PaddingRight: var(--pf-global--spacer--sm);
+  --pf-c-code-editor__tab--PaddingBottom: var(--pf-global--spacer--form-element);
+  --pf-c-code-editor__tab--PaddingLeft: var(--pf-global--spacer--sm);
+  --pf-c-code-editor__tab--before--BorderWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-code-editor__tab--before--BorderTopColor: var(--pf-global--BorderColor--100);
+  --pf-c-code-editor__tab--before--BorderRightColor: var(--pf-global--BorderColor--100);
+  --pf-c-code-editor__tab--before--BorderBottomColor: var(--pf-global--BackgroundColor--100);
+  --pf-c-code-editor__tab--before--BorderLeftColor: var(--pf-global--BorderColor--100);
+
+  width: 100%;
+}
+
+.pf-c-code-editor__control {
+  position: relative;
+  height: var(--pf-c-code-editor__control--Height);
```

Comment:
I'm not sure why you have the children of `__control` as `position: absolute`... that requires this `height`, and it also means that if the height of the children change for whatever reason, this `height` needs to change.

Couldn't this element just be a flex layout, with `align-items: flex-end`, and `margin-left: auto` on the tab element to push it to the right side? You shouldn't need a z-index on the children, either. You could just add `position: relative` to this element, the remove `position: relative` on the `__code` element. That will mean you should just use a regular border on the `__code` element instead of a pseudo element, but that's preferred anyways. Currently the border is overlapping the content, and you can't click into the `__code` element because the pseudo element is overlaid on top of it.

<img width="109" alt="Screen Shot 2020-09-08 at 5 58 54 PM" src="https://user-images.githubusercontent.com/35148959/92535558-1b952300-f1fd-11ea-8d95-b8eff19ecb3b.png">